mesonbuild/interpreter/interpreter.py

```diff
@@ -2211,67 +2208,68 @@ def func_install_subdir(self, node: mparser.BaseNode, args: T.Tuple[str],
         self.build.install_dirs.append(idir)
         return idir
 
-    @FeatureNewKwargs('configure_file', '0.47.0', ['copy', 'output_format', 'install_mode', 'encoding'])
-    @FeatureNewKwargs('configure_file', '0.46.0', ['format'])
-    @FeatureNewKwargs('configure_file', '0.41.0', ['capture'])
-    @FeatureNewKwargs('configure_file', '0.50.0', ['install'])
-    @FeatureNewKwargs('configure_file', '0.52.0', ['depfile'])
-    @FeatureNewKwargs('configure_file', '0.60.0', ['install_tag'])
-    @permittedKwargs({'input', 'output', 'configuration', 'command', 'copy', 'depfile',
-                      'install_dir', 'install_mode', 'capture', 'install', 'format',
-                      'output_format', 'encoding', 'install_tag'})
     @noPosargs
-    def func_configure_file(self, node, args, kwargs):
-        if 'output' not in kwargs:
-            raise InterpreterException('Required keyword argument "output" not defined.')
-        actions = {'configuration', 'command', 'copy'}.intersection(kwargs.keys())
-        if len(actions) == 0:
+    @typed_kwargs(
+        'configure_file',
+        KwargInfo('output', str, required=True),
+        KwargInfo('capture', bool, default=False, since='0.41.0'),
+        KwargInfo('format', str, default='meson', since='0.46.0',
+                  validator=in_set_validator({'meson', 'cmake', 'cmake@'})),
+        KwargInfo('output_format', str, default='c', since='0.47.0',
+                  validator=in_set_validator({'c', 'nasm'})),
+        DEPFILE_KW.evolve(since='0.52.0'),
+        # Cannot use shared implementation until None backwards compat is dropped
+        KwargInfo('install', (bool, NoneType), since='0.50.0'),
```

Comment:
Seems like install could become a constant KwargInfo like DEPFILE_KW above,

Reply:
Same with install_dir

Reply:
There is a comment describing why we cannot.

Reply:
I am blind

Reply:
There is a const version, but it only allows a bool, and configure_file has a legacy behavior when `install` isn't passed.